src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestExtractFileValidationServiceImpl.java

```diff
@@ -3,28 +3,168 @@
+    protected ConcurRequestedCashAdvanceService concurRequestedCashAdvanceService;
+    protected ConcurAccountValidationService concurAccountValidationService;
+    protected PersonService personService;
+    protected ConfigurationService configurationService;
+    protected ConcurBatchUtilityService concurBatchUtilityService;
+
+    public boolean requestExtractHeaderRowValidatesToFileContents(ConcurRequestExtractFile requestExtractFile) {
+        boolean headerValidationPassed;
+        LOG.info("batchDate=" + requestExtractFile.getBatchDate() + "=   recordCount=" + requestExtractFile.getRecordCount() + "=     totalApprovedAmount=" + requestExtractFile.getTotalApprovedAmount() + "=");
+        headerValidationPassed = (fileRowCountMatchesHeaderRowCount(requestExtractFile) &&
+                                  fileOnlyContainsOurEmployeeCustomerIndicator(requestExtractFile) &&
+                                  fileTotalApprovedAmountsMatchHeaderTotalApprovedAmount(requestExtractFile));
+        LOG.info("Header validation : " + ((headerValidationPassed) ? "PASSED" : "FAILED"));
+        return headerValidationPassed;
+    }
+
+    public void performRequestDetailLineValidation(ConcurRequestExtractRequestDetailFileLine detailFileLine) {
+        if (requestDetailFileLineIsCashAdvance(detailFileLine)) {
+            boolean lineValidationPassed;
+            lineValidationPassed = (requestedCashAdvanceHasNotBeenUsedInExpenseReport(detailFileLine) &&
+                                    requestedCashAdvanceIsNotBeingDuplicated(detailFileLine) &&
+                                    requestedCashAdvanceAccountingInformationIsValid(detailFileLine));
+            detailFileLine.getValidationResult().setValidCashAdvanceLine(lineValidationPassed);
+            LOG.info("Detail File Line validation : " + ((lineValidationPassed) ? "PASSED" : ("FAILED" + KFSConstants.NEWLINE + detailFileLine.toString())));
+        }
+    }
 
-    public boolean requestExtractHeaderRowValidatesToFileContents(List<ConcurRequestExtractFile> requestExtractFiles) {
-        boolean headerValidationPassed = false;
-        if (requestExtractFiles.isEmpty() || requestExtractFiles.size() != 1) {
-            LOG.error("Header validation should have received single parsed file. More or less than one parsed file was detected.");
+    /**
+     * A request detail line is considered a cash advance when request entry detail type is set to the cash advance indicator.
+     *
+     * @param detailFileLine
+     * @return boolean
+     */
+    private boolean requestDetailFileLineIsCashAdvance(ConcurRequestExtractRequestDetailFileLine detailFileLine) {
+        if (StringUtils.isNotEmpty(detailFileLine.getRequestEntryExpenseType()) &&
+            detailFileLine.getRequestEntryExpenseType().equalsIgnoreCase(ConcurConstants.CASH_ADVANCE_INDICATOR)) {
```

Comment:
You could simply this if statement by using StringUtils.equalsIgnoreCase(detailFileLine.getRequestEntryExpenseType(), (ConcurConstants.CASH_ADVANCE_INDICATOR)